libraries/src/Document/HtmlDocument.php

```diff
@@ -471,8 +472,17 @@ public function getBuffer($type = null, $name = null, $attribs = array())
 
 		if ($this->_caching == true && $type == 'modules' && $name !== 'debug')
 		{
-			$cache = \JFactory::getCache('com_modules', '');
-			$hash = md5(serialize(array($name, $attribs, null, get_class($renderer))));
+			$cache = CmsFactory::getCache('com_modules', '');
+			$hash = md5(
+				serialize(
+					array(
+						$name,
+						$attribs,
+						\get_class($renderer),
+						CmsFactory::getApplication()->input->get('Itemid')
```

Comment:
That is user provided content right? Can we please make sure we only go with an integer value here.

```suggestion
						CmsFactory::getApplication()->input->get('Itemid', 0, 'int')
```

Better would be to do an `$itemId = (int) CmsFactory::getApplication()->input->get('Itemid', 0, 'int');` above i guess.

Reply:
okay, I will change